### PR TITLE
Should we ensure `relative_hidden_states` to involve T_{f}^{+} and T_{f}^{-}?

### PR DESCRIPTION
**Pull request by novakboskov**
Reading the paper, I got the feeling that `relative_hidden_states[layer]` should be the difference between positive and negative function/concept templates. `RepReadingPipeline.get_directions` seemingly relies on `train_inputs[::2]` being the positive and `train_inputs[1::2]` the negative ones, but it does not ensure that. This may affect some downstream usages of `RepReadingPipeline.get_directions`.

For example, in `honesty.ipynb`, the training data gets prepared by `honesty_function_dataset`, which shuffles the labels of  prompts: 
https://github.com/andyzoujm/representation-engineering/blob/5455d8a375d5fb1cb191f9ebcd089b7c21e9a31e/examples/honesty/utils.py#L53-L57

When such prepared data is passed to `RepReadingPipeline.get_directions`, `train_inputs[::2]` are not necessarily the positive function/concept templates.

Am I reading this right?

### CODE REVIEW

**Comment by rgs2151** [comment]
I think you are right
